ProjNet/CoordinateSystems/Transformations/MathTransform.cs

```diff
@@ -1,246 +1,523 @@
+        /// <param name="outYs">A buffer for y-ordinate values</param>
+        /// <param name="strideX">A stride value for the x-ordinate series</param>
+        /// <param name="strideY">A stride value for the y-ordinate series</param>
+        /// <remarks>All series and buffers must not be <value>null</value></remarks>
+        /// <exception cref="ArgumentException">If the <paramref name="strideX"/> or <paramref name="strideY"/> values are <c>&lt;= 0</c></exception>
+        public void Transform(ReadOnlySpan<double> inXs, ReadOnlySpan<double> inYs, 
+            Span<double> outXs, Span<double> outYs, int strideX = 1, int strideY = 1)
+        {
+            if (strideX <= 0 || strideY <= 0)
+                throw new ArgumentException("Strides for x- and y- ordinates must be positive");
+
+            TransformCore(inXs, inYs, _inZs, outXs, outYs, _outZs, strideX, strideY, 0);
+        }
+
+        /// <summary>
+        /// Transforms a series of 2-dimensional points defined by their ordinates
+        /// </summary>
+        /// <param name="inXs">A series of x-ordinate values</param>
+        /// <param name="inYs">A series of y-ordinate values</param>
+        /// <param name="inZs">A series of z-ordinate values</param>
+        /// <param name="outXs">A buffer for x-ordinate values</param>
+        /// <param name="outYs">A buffer for y-ordinate values</param>
+        /// <param name="outZs">A buffer for y-ordinate values</param>
+        /// <param name="strideX">A stride value for the x-ordinate series</param>
+        /// <param name="strideY">A stride value for the y-ordinate series</param>
+        /// <param name="strideZ">A stride value for the z-ordinate series</param>
+        /// <remarks>All series and buffers must not be <value>null</value></remarks>
+        /// <exception cref="ArgumentException">If the <paramref name="strideX"/> or <paramref name="strideY"/> values are <c>&lt;= 0</c></exception>
+        public void Transform(ReadOnlySpan<double> inXs, ReadOnlySpan<double> inYs, ReadOnlySpan<double> inZs,
+            Span<double> outXs, Span<double> outYs, Span<double> outZs, int strideX = 1, int strideY = 1, int strideZ = 1)
+        {
+            if (strideX <= 0 || strideY <= 0)
+                throw new ArgumentException("Strides for x- and y- ordinates must be positive");
+
+            // check span sizes
+            int inXElements = inXs.Length / strideX;
+            if (strideX > 1) inXElements += 1;
+            int inYElements = inYs.Length / strideY;
+            if (strideY > 1) inYElements += 1;
+
+            if (inXElements != inYElements ||
+                inXs.Length != outXs.Length ||
+                inYs.Length != outYs.Length)
+            {
+                throw new ArgumentException("Provided x- and y-spans must be of the same length.");
+            }
+
+            // If a stride value for z is positive we must check this, too
+            if (strideZ > 0 && inZs.Length > 0)
```

Comment:
The problem with `default` is that it provides a zero-length span that cannot be addressed.

Reply:
> The problem with `default` is that it provides a zero-length span that cannot be indexed.

Sure, but that's a problem that we can solve: if the user gives us a span that `.IsEmpty`, then we replace it in this method with our clever one-element span for the sake of the inner method.  Otherwise, we validate that it's the same length.